tests/test_api.py

```diff
@@ -162,50 +159,46 @@ def test_read_write_read_compare(self):
 
 
     def test_sign_verify(self):
+        root_path = os.path.join(self.repo_dir, 'metadata', 'root.json')
+        root:Root = Metadata.from_file(root_path).signed
+
+        # Locate the public keys we need from root
+        targets_keyid = next(iter(root.roles["targets"].keyids))
+        targets_key = root.keys[targets_keyid]
```

Comment:
I noticed now this line in the spec saying

> ... Clients MUST calculate each KEYID to verify this is correct for the associated key ...

Calculating the `KEYID` is now done during `sslib.format_metadata_to_key` but is never compared to the one read from the metadata files. I can't decide is there a benefit in explicitly doing this check. Or is it done somewhere and I am missing it?

Reply:
* it's not done anywhere
* I've read through the sslib implementation and I don't think the keyid is actually reproducible -- things like securesystemslib.settings.HASH_ALGORITHMS value affect it.
* I think specifying how the unique key must be created in the spec is a mistake: it serves no purpose. The spec should just demand the keyid uniquely identifies the key within this metadata file
* I also think that's how we should handle keyid in python-tuf, even if spec is what it is: IMO we can't do anything else

Reply:
see also https://github.com/theupdateframework/taps/blob/master/tap12.md

Reply:
Thanks for the explanation. Feel like I am convinced now!

Reply:
> I noticed now this line in the spec saying
> 
> > ... Clients MUST calculate each KEYID to verify this is correct for the associated key ...
> 
> Calculating the `KEYID` is now done during `sslib.format_metadata_to_key` but is never compared to the one read from the metadata files. I can't decide is there a benefit in explicitly doing this check. Or is it done somewhere and I am missing it?

FYI https://github.com/theupdateframework/tuf/pull/1014, specifically https://github.com/theupdateframework/tuf/commit/9fa92e22865b2ba4e164493564042cf3ca778bee